File: 프로그래머스/찾아라 프로그래밍 마에스터/게임 맵 최단거리.py
# 게임 맵 최단거리
from collections import deque
dx = [1,-1,0,0]
dy = [0,0,1,-1]
def solution(maps):
    answer = 0
    n, m = len(maps), len(maps[0])
    queue = deque()
    queue.append(([0,0]))
    maps[0][0] = 2
    while queue:
        cur = queue.popleft()
        for i in range(4):
            nx,ny = cur[0] + dx[i] , cur[1] + dy[i]
            if 0 <= nx < n and 0 <= ny < m and maps[nx][ny] == 1:
                maps[nx][ny] = maps[cur[0]][cur[1]] + 1
                queue.append([nx,ny])
    return maps[n-1][m-1] -1 if maps[n-1][m-1] > 1 else -1

File: 프로그래머스/찾아라 프로그래밍 마에스터/게임 맵 최단거리.js
// 게임 맵 최단거리
function solution(maps) {
    let answer = 0;
    const dx = [0,0,1,-1], dy = [1,-1,0,0];
    let queue = [];
    const n = maps.length, m = maps[0].length;
    maps[0][0] = 2;
    queue.push([0,0]);
    while(queue.length > 0){
        let cur = queue.shift();
        for(let i = 0; i < 4; i++){
            const nx = cur[0] + dx[i], ny = cur[1] + dy[i];
            if(nx >= 0 && nx < n && ny >= 0 && ny < m && maps[nx][ny] == 1){
                queue.push([nx,ny]);
                maps[nx][ny] = maps[cur[0]][cur[1]] + 1;
            }
        }
    }
    return maps[n-1][m-1] > 1 ? maps[n-1][m-1]-1 : -1;
}